packages/go/bomenc/normalize.go

```diff
@@ -0,0 +1,170 @@
+// Copyright 2024 Specter Ops, Inc.
+//
+// Licensed under the Apache License, Version 2.0
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package bomenc
+
+import (
+	"bytes"
+	"errors"
+	"io"
+	"unicode/utf8"
+)
+
+// Normalizer interface defines the methods for accessing normalized content.
+// This allows consumers of the package to work with normalized data without
+// knowing the specifics of how the normalization was performed.
+type Normalizer interface {
```

Comment:
If there's only one implementation do we need this interface?

Reply:
@zinic not necessarily, but I wanted to convey that the function returning this interface is meant read-only.

Reply:
:+1: